paddle/framework/net_design.md

```diff
@@ -0,0 +1,251 @@
+# Network Design
+
+`Network` is the container and controller of a set of operators,
+user can build a real network from a `NetDef` which is a protobuf message 
+and use `Network.Run()` to run all the operators in the network.
+
+The `Network` will
+
+- manage all the operators contained in the network.
+- not own any `Variable`.
+
+# API
+
+## NetBase
+To make the `Network` extendable, a base class is defined like this
+
+```c++
+// operator's index stored in a network.
+typedef int OpIndex;
+
+// The minimum a network should be implemented.
+class NetBase {
```

Comment:
NetBase ==> Net, because our other base classes, including `Operator`, don't have the `Base` suffix. Let's unify the style.